src/EditorFeatures/Core.Wpf/NavigateTo/NavigateToItemProvider.cs

```diff
@@ -112,7 +113,18 @@ private void StartSearch(INavigateToCallback callback, string searchValue, IImmu
                 kinds,
                 _threadingContext.DisposalToken);
 
-            _ = searcher.SearchAsync(searchCurrentDocument, _cancellationTokenSource.Token);
+            _ = SearchAsync(searcher, searchCurrentDocument, _cancellationTokenSource.Token);
+        }
+
+        private async Task SearchAsync(NavigateToSearcher searcher, bool searchCurrentDocument, CancellationToken cancellationToken)
```

Comment:
❗ This is unnecessary (the caller is fire-and-forget so catching exceptions is meaningless)

Reply:
📝 Even with the rewrite to add an exception filter, you can replace the separate method with `.ReportNonFatalErrorUnlessCancelledAsync(cancellationToken)` in the caller.

Reply:
> This is unnecessary (the caller is fire-and-forget so catching exceptions is meaningless)

What about the unhandled task fault failure handler?

Reply:
It was disabled by default [in .NET 4.5](https://devblogs.microsoft.com/pfxteam/task-exception-handling-in-net-4-5), and removed completely in https://github.com/dotnet/coreclr/pull/18783.

Reply:
updated!